### PR TITLE
add terrain generation with open simplex noise

### PR DESCRIPTION
**Pull request by KarolinePlum**
progress on issue #50

### CODE REVIEW

File: base/src/gen/world/mod.rs [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
 use world::{CHUNK_SIZE, Chunk, ChunkIndex, ChunkProvider, HeightType, HexPillar};
 use world::{GroundMaterial, PillarSection, Prop, PropType};
 use rand::Rand;
 use gen::PlantGenerator;
+use noise::{PermutationTable, open_simplex2};
+use gen::seeded_rng;
+
+// pub const LANDSCAPE_FLAT: f32 = 0.01;
+pub const LANDSCAPE_MODERATE: f32 = 0.05;
+// pub const LANDSCAPE_HILLY: f32 = 0.1125;
```
Comment (LukasKalbertodt) [inline base/src/gen/world/mod.rs]: Do you need those constants later? If not, please remove the comments ;)

File: base/src/gen/world/mod.rs [hunk 2]
```diff
@@ -32,12 +47,15 @@ impl ChunkProvider for WorldGenerator {
...
 
+                let noise = open_simplex2::<f32>(&self.terrain_table,
+                                                 &[(i as f32) * LANDSCAPE_MODERATE,
+                                                   (j as f32) * LANDSCAPE_MODERATE]);
+                let height = 50.0 + noise * 50.0;
+                println!("{}", height);
                 let ground_section = PillarSection::new(GroundMaterial::Dirt,
                                                         HeightType::from_units(0),
-                                                        HeightType::from_units(height));
+                                                        HeightType::from_units(height as u16));
```
Comment (LukasKalbertodt) [inline base/src/gen/world/mod.rs]: I think you can use `height.into()` here, right?

File: base/src/gen/world/mod.rs [hunk 3]
```diff
@@ -47,7 +65,7 @@ impl ChunkProvider for WorldGenerator {
                     let gen = PlantGenerator::rand(&mut rng);
 
                     props.push(Prop {
-                        baseline: HeightType::from_units(height),
+                        baseline: HeightType::from_units(height as u16),
```
Comment (LukasKalbertodt) [inline base/src/gen/world/mod.rs]: Same as above.

File: base/src/gen/world/mod.rs (hunk 1 above)
Comment (KarolinePlum) [inline base/src/gen/world/mod.rs]: Yes we might need them later for generating biomes

File: base/src/gen/world/mod.rs [hunk 4]
```diff
@@ -1,21 +1,36 @@
 //! Procedurally generating the game world.
 //!
 
-use world::{CHUNK_SIZE, Chunk, ChunkIndex, ChunkProvider, HeightType, HexPillar};
+// extern crate noise;
```
Comment (jonas-schievink) [inline base/src/gen/world/mod.rs]: Unnecessary comment

File: base/src/world/chunk.rs [hunk 5]
```diff
@@ -54,12 +54,15 @@ impl Chunk {
     }
 
     /// Creates a `Chunk` using individual pillars returned by a closure
-    pub fn with_pillars<F>(mut func: F) -> Chunk
+    pub fn with_pillars<F>(chunk_index: ChunkIndex, mut func: F) -> Chunk
```
Comment (jonas-schievink) [inline base/src/world/chunk.rs]: I'm fine with this personally, but you could also do this calculation in the closure you pass to this

**Pull request merged**